diff --git a/src/preset/setup.js b/src/preset/setup.js
index 488a733727d4c4efc1ba649d3f4a6be5048fc5ab..4241e8c8b4c343ee504dfda071f0684f72d05cfe 100644
--- a/src/preset/setup.js
+++ b/src/preset/setup.js
@@ -7,7 +7,8 @@
 const findUp = require('find-up');
 const merge = require('lodash/merge');
 const path = require('path');
-const mockNativeModules = require('react-native/Libraries/BatchedBridge/NativeModules').default;
+const _mockNativeModules = require('react-native/Libraries/BatchedBridge/NativeModules');
+const mockNativeModules = _mockNativeModules.default || _mockNativeModules;
 const stackTrace = require('stacktrace-js');
 
 const publicExpoModules = require('./moduleMocks/expoModules');
@@ -296,6 +297,16 @@ try {
   }
 }
 
+// Mock missing TurboModules for RN 0.77+
+jest.mock('react-native/src/private/specs/modules/NativeSourceCode', () => ({
+  __esModule: true,
+  default: {
+    getConstants: () => ({
+      scriptURL: 'http://localhost:8081/index.bundle',
+    }),
+  },
+}));
+
 // Installs web implementations of global things that are normally installed through JSI.
 require('expo-modules-core/src/web/index.web');
 
